lib/backend.js

```diff
@@ -661,24 +661,33 @@ Backend.prototype.querySubscribe = function(agent, index, query, options, callba
         'DB does not support subscribe'
       ));
     }
-    backend.pubsub.subscribe(request.channel, function(err, stream) {
-      if (err) return callback(err);
-      if (options.ids) {
-        var queryEmitter = new QueryEmitter(request, stream, options.ids);
-        backend.emit('timing', 'querySubscribe.reconnect', Date.now() - start, request);
-        callback(null, queryEmitter);
-        return;
-      }
-      // Issue query on db to get our initial results
-      backend._query(agent, request, function(err, snapshots, extra) {
-        if (err) {
-          stream.destroy();
-          return callback(err);
+
+    var channels = [request.channel];
+
+    if (request.channels.length) {
+      channels = request.channels;
+    }
+
+    channels.forEach(function(channel) {
+      backend.pubsub.subscribe(channel, function(err, stream) {
+        if (err) return callback(err);
+        if (options.ids) {
+          var queryEmitter = new QueryEmitter(request, stream, options.ids);
+          backend.emit('timing', 'querySubscribe.reconnect', Date.now() - start, request);
+          callback(null, queryEmitter);
+          return;
         }
-        var ids = pluckIds(snapshots);
-        var queryEmitter = new QueryEmitter(request, stream, ids, extra);
-        backend.emit('timing', 'querySubscribe.initial', Date.now() - start, request);
-        callback(null, queryEmitter, snapshots, extra);
+        // Issue query on db to get our initial results
+        backend._query(agent, request, function(err, snapshots, extra) {
```

Comment:
Would be nice if we could avoid hitting the DB with the same query multiple times